jmespath.js

```diff
@@ -392,7 +392,14 @@
               }
               this.current = current;
           }
-          var literalString = stream.slice(start, this.current).trimLeft();
+          if (typeof String.prototype.trimLeft !== "function") {
```

Comment:
Given this check applies on a per-browser basis, I think it makes sense to move this logic to the top level scope so it's only defined once.  Otherwise every time `consumeLiteral` is called, we're recreating this function unnecessarily.